bioptim/dynamics/configure_problem.py

```diff
@@ -136,12 +139,30 @@ def torque_driven(
         ConfigureProblem.configure_qdot(nlp, True, False)
         ConfigureProblem.configure_tau(nlp, False, True, fatigue)
 
-        if implicit_dynamics:
+        if semi_implicit_dynamics:
```

Comment:
Since implicit_dynamics and semi_implicit_dynamics (and for the matter explicit_dynamics) are mutually exclusive, this should be an actual ENUM (something like: DynamicsType [but I think this is already taken]).
So here, you can test against this enum. This will make the API more explicit while preventing from someone setting them all to `True` (and not requiring a bunch of error catching)

Reply:
What about using a list of dynamics, in case I want explicit muscle_dynamics and implicit soft_contact_dynamics but semi_implicit_multibody_dynamics ? and pass all of them through an argument dynamics.

Reply:
I do not see a problem with that, as long as it is not a bunch of  mutually exclusive bool :)